### ISSUE TITLE
[TAStudio] Certian error window prevents bizhawk from closing.

### ISSUE THREAD

**Issue by EZGames69** [TASEmulators/BizHawk#2332]
I mentioned an error in this ticket: https://github.com/TASVideos/BizHawk/issues/2326

The issue in that ticket wasnt deemed as a priority, however, the error window is in itself an error.  That being this: 

![image](https://user-images.githubusercontent.com/44628443/91103994-d12b7800-e63a-11ea-953e-b86dbf72c774.png)

This window will keep popping up every time you try to press the X button or hit "ok", and it prevents you from being able to close that instance of bizhawk.  The only way to close it is to restart your computer.

**Comment by RetroEdit** [comment]
I doubt the only way is to restart your computer. You can probably close BizHawk through task manager (though feel free to correct me if I'm mistaken).

**Comment by EZGames69** [comment]
You are correct that it does close bizhawk, but I don't think this should be the only way to get around it.

**Comment by nattthebear** [comment]
We in general need to get rid of all "retry" mechanisms on unhandled exception handlers.  If something is universally catching all exceptions from deep calls not directly related to it, the only thing it can safely do is kill the application, possibly with an opportunity for error reporting first.  It's never safe to continue because you don't know what postconditions weren't satisfied and what objects are left in a dirty state.

**Comment by zeromus** [comment]
Yeah. Anyway, the window getting stuck open is not a whole lot different than the application closing. Application closing universally as you suggested should be considered a feature. 

THIS bug won't be fixed without fixing the underlying bug (the text will continue to get queried). Yoshirulz will find this out and then close it again. I suggest reopening a new request with natt's idea

**Comment by YoshiRulz** [comment]
I think closing TAStudio should prevent any problems, EmuHawk doesn't need to close entirely.

**Comment by nattthebear** [comment]
> I suggest reopening a new request with natt's idea

Yeah, I'm more musing about a perfect world; there's a reason why I didn't open a bug for my gripe

**Comment by zeromus** [comment]
We don't need to close tastudio. The logic you add to close tastudio will be more garbage in the way. If you added a once and for all close emuhawk method, the garbage won't proliferate. What does it matter if bizhawk stays open, if tastudio goes down? How much progress is the user gonna lose anyway? Nil, he wiped everything before trying to open the old tas. We're doing the users a favor tearing down emuhawk, at least whatever they do next will be in a known good state.

**Comment by YoshiRulz** [comment]
That makes sense, but I don't see how closing EmuHawk would be easier. Can't we just call `this.Close()` after the MessageBox here?
https://github.com/TASVideos/BizHawk/blob/956b71060e5dbbc9d045cc7d20973a5e3bf24c59/src/BizHawk.Client.EmuHawk/tools/TAStudio/TAStudio.ListView.cs#L370-L374

**Comment by zeromus** [comment]
It's not supposed to be easier, it's supposed to be a universal solution to uncaught exceptions at the top of a thread. 

Closing a window from a callback on a control on the window doesnt seem smart to me. Those query calls come from win32 hacks, it's not a normal "close the dialog in response to file > close" scenario. 

The window is probably then going to nag the user to save, or something, which will flop due to it being confused in the first place.

I guess what you propose is one line of garbage, but if it multiplies please GIVE UP
